### PR TITLE
[fix bug 1431412] Update GA script.

### PR DESCRIPTION
**Pull request by jpetto**
In the interest of keeping review on this PR sane, I'm going to split it into (at least) two commits.

1. Adding the new GA script and product tagging.
2. Adding `data-*` attributes all over the place.

@glogiotatidis - Can you give this first commit a review? Not being familiar with SUMO, I want to make sure I've got the new GA script on all necessary templates. It would also be nice if we had a public comment somewhere detailing our data collection practices (as we have for [mozilla.org](https://bugzilla.mozilla.org/show_bug.cgi?id=1122305#c8)).

### CODE REVIEW

File: kitsune/sumo/static/sumo/js/analytics.js [hunk 1]
```diff
@@ -1,119 +0,0 @@
...
-        _gaq.push(['_trackEvent',
-                   'Landing Page Read - 5 seconds',
-                   $('body').data('product-slug')]);
-      }
-    }, 5000);
-  }
-}
-
-
-function trackEvent(category, action, value) {
```
Comment (glogiotatidis) [inline kitsune/sumo/static/sumo/js/analytics.js]: What's the plan with pages calling this and `trackPageview` functions?

File: kitsune/settings.py [hunk 2]
```diff
@@ -911,6 +911,7 @@ def JINJA_CONFIG():
 GA_ACCOUNT = config('GA_ACCOUNT', 'something@developer.gserviceaccount.com')  # Google API Service Account email address
 GA_PROFILE_ID = config('GA_PROFILE_ID', default='12345678')  # Google Analytics profile id for SUMO prod
```
Comment (glogiotatidis) [inline kitsune/settings.py]: This is used in [`googleanalytics.py`](https://github.com/mozilla/kitsune/blob/master/kitsune/sumo/googleanalytics.py) which pulls GA stats into Kitsune. 

Is GTM compatible with this? In other other words will it populate the same tables / properties (or how they are called in GA)?

Comment (glogiotatidis) [reply kitsune/settings.py]: Just read your comment in #3027 . Will continue the discussion there.

File: kitsune/sumo/static/sumo/js/analytics.js [hunk 3]
```diff
@@ -1,119 +0,0 @@
...
-  if (_gaq) {
-    _gaq.push(['_trackEvent', category, action, value]);
-  }
-}
-
-function trackPageview(value) {
```
Comment (bensternthal) [inline kitsune/sumo/static/sumo/js/analytics.js]: It looks like some of these functions like trackPageview are used in quite a few places. I think we should think about migrating the few custom functions here or if the new ga will collect this data, deprecate and remove.

Comment (jpetto) [reply kitsune/sumo/static/sumo/js/analytics.js]: Yeah, this was going to be a sub-task of commit 2 (adding `data-*` attrs) referenced above - removing all old GA calls. I'll update my commit here to keep this file.

Comment (jpetto) [reply kitsune/sumo/static/sumo/js/analytics.js]: > I think we should think about migrating the few custom functions here or if the new ga will collect this data, deprecate and remove.

The intent (as I see it) is to remove *all* of the calls to `trackPageview` and `trackEvent` and replace with `data-*` attributes as requested in the PDF. However, there may be some gaps left, which we'd only learn by going through the process. In other words, there may be questions for AP when removing current tracking calls and adding new `data-*` attributes.

**Comment by bensternthal** [comment]
@jpetto ok before I go down editing a bunch of stuff can you sanity check the todo list for the next part of the pr:

- [ ] Find and update all uses of ```a[data-ga-click]```
- [ ] Find and update all uses of ```button[data-ga-click]```
- [ ] Find and update all uses of function ```trackevent```
- [ ] Find and update all uses of function```trackPageview```
- [ ] Add the new tracking AP specified in their doc.

Two clarifications

1. Can you verify if [this](https://docs.google.com/presentation/d/1UEUXYVMXD7w12YL7Io4SxkJ4CGd3-v3ewMN-A6kOH2s/edit#slide=id.g2b8b5c67bb_0_11) is the spec to use when specifying html data attributes?
2. When I replace the js for trackevent and trackPageview, can you give me the format to use including the function call, example ```ga('send', 'event', '<category>', '<action>', '<label>');```

Outside of that I think we need to keep/update the [code](https://github.com/mozilla/kitsune/blob/master/kitsune/sumo/static/sumo/js/analytics.js#L88) that tracks reads. I dont see a way of accomplishing this without the custom code that does the time outs.

**Comment by jpetto** [comment]
A few notes on the list:

- I see some elements with `data-ga-click` attributes that would not be covered under the requirements set by AP (e.g. the form in `new_question.html`). You may want to double check the removal of these with AP.
- For each removal of a `trackEvent` call, make sure that the new GTM implementation will (via `data-*` attributes) continue to gather the same data, or verify the data capture being removed is no longer needed.
- The `trackPageview` calls appear to be a little different, as they seem to solely be tied to searches. It may be worth asking AP how to handle that data collection with GTM. (If anything, it would probably be a JS call on our end to `gtag` with specific arguments.)

In terms of the clarifications:

1. The data attributes in the spec are just (solid) suggestions that we could tweak if we wanted to. I think what they've provided is fine. (For example, we could use `data-action` instead of `data-event-action` to save on file size/typing. We'd just need to tell AP which attribute names we've picked.)
2. I don't think you need to do any JS replacement - only removal. On moz.org, for basic tracking, all we have are `data-*` attributes. The JS loaded from GTM sees those attributes and makes the call to `gatag` for you. The only times we make specific JS calls on moz.org are for non-basic interactions like playing/pausing a video or scrolling down a page.

In terms of the code that tracks reads, I think this again should be run by AP/the Mozilla analytics person handling this work. Maybe they've decided that data isn't important? Or, if we want/need to keep it, it's again probably just a JS call on our end to `gtag` with specific arguments.

The way I am reading the spec from AP is that it covers the most important data to be captured. We may discover gaps or introduce new things we want to track, in which case we'd either add more `data-*` attributes or write some custom JS on our end.

**Comment by glogiotatidis** [comment]
Replaced by #3076

**Pull request closed without merging**